pishutdown.py

```diff
@@ -23,14 +27,17 @@ def buttonStateChanged(pin):
 
     if not (GPIO.input(pin)):
         # button is down
-        buttonPressedTime = datetime.now()
+        if buttonPressedTime is None:
```

Comment:
I would not add this line. I think adding it may cause a button press to not register, e.g. when signal bounces **on**, **off**, **on**. Without the condition I think the worst that can happen if bouncing is that `buttonPressedTime` gets set  again a few milliseconds later, which should not matter when we are talking about seconds long button presses.